code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,177 @@
+ * such as [AepText], [AepImage], [AepButton], and [AepDismissButton], as well as creating
+ * the corresponding UI templates like [SmallImageTemplate].
+ */
+internal object ContentCardSchemaDataUtils {
+    /**
+     * Creates an instance of [AepText] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepText].
+     * @return An instance of [AepText].
+     */
+    private fun createAepText(map: Map<String, Any>): AepText {
+        return AepText(
+            content = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.CONTENT, "")
+        )
+    }
+
+    /**
+     * Creates an instance of [AepImage] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepImage].
+     * @return An instance of [AepImage].
+     */
+    private fun createAepImage(map: Map<String, Any>): AepImage {
+        return AepImage(
+            url = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.URL, null),
+            darkUrl = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.DARK_URL, null)
+        )
+    }
+
+    /**
+     * Creates an instance of [AepButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepButton].
+     * @return An instance of [AepButton].
+     */
+    private fun createAepButton(map: Map<String, Any>): AepButton {
+        return AepButton(
+            id = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.INTERACT_ID, null),
+            actionUrl = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.ACTION_URL, null),
+            text = DataReader.optTypedMap(Any::class.java, map, MessagingConstants.ContentCard.UIKeys.TEXT, null).let { createAepText(it) }
+        )
+    }
+
+    /**
+     * Creates an instance of [AepDismissButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepDismissButton].
+     * @return An instance of [AepDismissButton].
+     */
+    private fun createAepDismissButton(map: Map<String, Any>): AepDismissButton {
+        return AepDismissButton(
+            style = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.STYLE, MessagingConstants.ContentCard.UIKeys.NONE)
+        )
+    }
+
+    /**
+     * Parses the given [ContentCardSchemaData] to create an [AepUITemplate].
+     *
+     * @param contentCardSchemaData The content card schema data containing the content map.
+     * @return The parsed [AepUITemplate] or null if parsing fails.
+     */
+    internal fun getTemplateModel(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+        try {
+            val contentMap =
+                contentCardSchemaData.content as? Map<String, Any> ?: throw IllegalArgumentException("Content map is null")
+            val id = DataReader.getString(contentMap, MessagingConstants.ContentCard.UIKeys.ID)
```

Comment:
There is not id field send from the server. This needs to be the proposition id.
`contentCardSchemaData.parent.proposition.uniqueId`